website/docs/guides/delivery.mdx

```diff
@@ -0,0 +1,473 @@
+
+```yaml title="flagger/kustomization.yaml"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: flagger
+resources:
+- namespace.yaml
+- source.yaml
+- releases.yaml
+```
+
+## Installing Linkerd using Flux
+
+For the Linkerd installation, a Kustomization file will be used.
+
+In order to support mTLS connections between meshed pods, Linkerd requires a trust anchor certificate and an issuer certificate with its corresponding key. These certificates are automatically created when the `linkerd install` command is used but when using a Helm chart to install Linkerd, these certificates need to be provided.
+
+Follow [these instructions](https://linkerd.io/2.11/tasks/generate-certificates/#generating-the-certificates-with-step) to generate the certificates using the `step` CLI.
+
+To generate the trust anchor certificate run:
+```bash
+step certificate create root.linkerd.cluster.local ca.crt ca.key \
+--profile root-ca --no-password --insecure
+```
+
+To generate the issuer certificate run:
+```bash
+step certificate create identity.linkerd.cluster.local issuer.crt issuer.key \
+--profile intermediate-ca --not-after 8760h --no-password --insecure \
+--ca ca.crt --ca-key ca.key
+```
+
+Add the `ca.crt`, `issuer.crt` and `issuer.key` files to the cluster repository under a `linkerd` directory.
+
+To control where the Linkerd components get installed, we need to add a Namespace resource:
+
+```yaml title="linkerd/namespace.yaml"
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: linkerd
+  labels:
+    config.linkerd.io/admission-webhooks: disabled
+```
+
+Make the Linkerd Helm repository available in the cluster, by adding the following manifest to the cluster repository:
+
+```yaml title="linkerd/source.yaml"
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: linkerd
+spec:
+  interval: 1h
+  url: https://helm.linkerd.io/stable
+```
+
+Then, to install the latest version of Linkerd, add the following `HelmRelease` manifests to the cluster repository:
+
+```yaml title="linkerd/releases.yaml"
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: linkerd
+spec:
+  interval: 10m
+  chart:
+    spec:
+      chart: linkerd2
+      reconcileStrategy: ChartVersion
+      sourceRef:
+        kind: HelmRepository
+        name: linkerd
+  install:
+    crds: Create
+  upgrade:
+    crds: CreateReplace
+  valuesFrom:
+    - kind: Secret
+      name: linkerd-certs
+      valuesKey: ca.crt
+      targetPath: identityTrustAnchorsPEM
+    - kind: Secret
+      name: linkerd-certs
+      valuesKey: issuer.crt
+      targetPath: identity.issuer.tls.crtPEM
+    - kind: Secret
+      name: linkerd-certs
+      valuesKey: issuer.key
+      targetPath: identity.issuer.tls.keyPEM
+  values:
+    installNamespace: false
+    identity:
+      issuer:
+        crtExpiry: "2023-07-18T20:00:00Z"
```

Comment:
Maybe add a comment along the lines of #change this value ?

Reply:
Yes ideally this needs to be set to the expiry date of the generated certificate. I'll add a comment.